### PR TITLE
Solr 9

### PR DESCRIPTION
**Pull request by kdebski85**
Please first create rohitbemax:branch_9x branch, so I can change target branch.

### CODE REVIEW

File: src/main/java/org/apache/solr/handler/dataimport/DocBuilder.java [hunk 1]
```diff
@@ -114,8 +114,10 @@ private VariableResolver getVariableResolver() {
       VariableResolver resolver = null;
       String epoch = propWriter.convertDateToString(EPOCH);
       if(dataImporter != null && dataImporter.getCore() != null
-          && dataImporter.getCore().getResourceLoader().getCoreProperties() != null){
-        resolver =  new VariableResolver(dataImporter.getCore().getResourceLoader().getCoreProperties());
+              && dataImporter.getCore().getCoreContainer() != null
+              && dataImporter.getCore().getCoreContainer().getConfig() != null
+              && dataImporter.getCore().getCoreContainer().getConfig().getSolrProperties() != null) {
```
Comment (kdebski85) [inline src/main/java/org/apache/solr/handler/dataimport/DocBuilder.java:119]: note: I am not 100% sure if these are correct properties. I found properties in a few different places.

File: src/main/java/org/apache/solr/handler/dataimport/CryptoKeys.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import java.security.MessageDigest;
+import java.util.Arrays;
+import java.util.Base64;
+
+import org.apache.solr.common.SolrException;
+
+/**
+ * Code copied from Solr 8.11.1
+ */
+public class CryptoKeys {
```
Comment (kdebski85) [inline src/main/java/org/apache/solr/handler/dataimport/CryptoKeys.java:19]: note: `decodeAES` method is used in `JdbcDataSource` class. It was removed from Solr 9, so I copied it here from 8.11.1.

File: src/main/java/org/apache/solr/handler/dataimport/SolrPaths.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Code copied from Solr 8.11.1
+ */
+public class SolrPaths {
```
Comment (kdebski85) [inline src/main/java/org/apache/solr/handler/dataimport/SolrPaths.java:19]: note: `locateSolrHome` method is used in `SimplePropertiesWriter` class. It was removed from Solr 9, so I copied it here from 8.11.1.

**Comment by kdebski85** [comment]
@epugh Could you please create `rohitbemax:branch_9x` branch, so I could change the target branch of this pull request? It should not be merged to branch_8x.
Could you please review this PR?

**Comment by chatman** [comment]
Done, created branch_9x. Thanks for your PR, lets have it merged soon :-)

**Comment by kdebski85** [comment]
Thanks! I changed the target branch to `branch_9x`

**Comment by kdebski85** [comment]
Please review my changes

**Comment by kdebski85** [comment]
@chatman In the meantime, I changed my job and I no longer use Solr (at least not directly). If you think any changes should be done to this PR, I could try to do them, but I will not be able to test on production-like data. Feel free to make changes to this PR.

**Comment by aboodalfar** [comment]
@kdebski85 @shalinmangar Hi , Please merge this request please please

**Comment by aboodalfar** [comment]
hi any update ?

**Comment by kdebski85** [comment]
@aboodalfar Have you reviewed and tested these changes?

**Comment by aboodalfar** [comment]
> @aboodalfar Have you reviewed and tested these changes?

no , I just need this update to support version 9

**Comment by idurgakalyan** [comment]
Hi, Any update on this?

**Comment by chatman** [comment]
Thank you for your patience and apologies for the delay. I shall get to this PR this week and attempt to release a 9x version of DIH. Thanks @kdebski85 @idurgakalyan !

**Comment by idurgakalyan** [comment]
Hi, Any update on testing and merging?

**Comment by chatman** [comment]
The project now has a new home and I just assumed ownership of the project under the SearchScale organization. https://lists.apache.org/thread/bmovqmy9gsg6fcfyl2oj6v2lft3x0jpq

I shall attempt to have a release done as soon as possible. Thank you all for the patience and understanding.

**Comment by chatman** [comment]
Thanks @kdebski85, I tested it with few records into a MariaDB table, and it worked flawlessly. Made a release for Solr 9.0.

However, when I compiled it with Solr 9.2, there were some compilation errors. I think we need another PR (branch_9x) and a release for Solr 9.2.

If you are interested in upgrading DIH to 9.2, I can invite you as a maintainer and share the release process with you.

**Pull request merged**